examples/reinforcement_learning/actor_critic_cartpole.py

```diff
@@ -0,0 +1,119 @@
+"""
+Title: Actor Critic Method
+Author: [Apoorv Nandan](https://twitter.com/NandanApoorv)
+Date created: 2020/05/13
+Last modified: 2020/05/13
+Description: Implement Actor Critic Method in CartPole environment
+"""
+"""
+## Setup
+"""
+
+import gym
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow import math
+from tensorflow.keras import losses
+from tensorflow.keras import layers
+from tensorflow.keras import optimizers
+
+seed = 543
+gamma = 0.99
+env = gym.make("CartPole-v0")
+env.seed(seed)
+eps = np.finfo(np.float32).eps.item()
+
+"""
+## Implement Actor Critic network
+"""
+
+num_inputs = 4
+num_actions = 2
+num_hidden = 128
+
+inputs = layers.Input(shape=(num_inputs,))
+common = layers.Dense(num_hidden, activation="relu")(inputs)
+action = layers.Dense(num_actions, activation="softmax")(common)
+critic = layers.Dense(1)(common)
+
+model = keras.Model(inputs=inputs, outputs=[action, critic])
+
+"""
+## Train
+"""
+
+optimizer = optimizers.Adam(learning_rate=0.01)
+action_probs_history = []
+critic_value_history = []
+rewards_history = []
+running_reward = 0
+episode_count = 0
+
+while True:  # run till solved
+    state = env.reset()
+    episode_reward = 0
+    with tf.GradientTape() as tape:  # keep track of ops for backprop
+        for timestep in range(1, 10000):
+            state = tf.convert_to_tensor(state)
+            state = tf.expand_dims(state, 0)
+
+            action_probs, critic_value = model(state)
```

Comment:
Also it would be more performant to put the 3 lines above in a `tf.function`

Reply:
Tested this. 

@tf.function takes 0.0005 seconds on an average.
The current code takes 0.0009 seconds on an average.

Both these numbers are very low compared to the overall time taken per episode, which varies from 0.01 to 0.06 seconds on an average.

Reply:
Ok, we can leave it as is then.